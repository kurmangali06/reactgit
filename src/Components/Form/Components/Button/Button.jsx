import React from "react";

export const Button = ({ disabled, onButtonClick }) => (
  <button type="submit"
    className="submit" disabled={disabled} onClick={onButtonClick} >
    click
  </button>
);