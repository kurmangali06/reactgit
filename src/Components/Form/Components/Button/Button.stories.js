"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Disabled = exports.Primary = void 0;
const react_1 = __importDefault(require("react"));
const Button_1 = require("./Button");
exports.default = {
    title: 'MyComponents/Button',
    component: Button_1.Button,
    argTypes: {
        onButtonClick: { action: 'click' },
    },
};
const Template = (args) => react_1.default.createElement(Button_1.Button, Object.assign({}, args));
exports.Primary = Template.bind({});
exports.Primary.args = {
    disabled: false,
};
exports.Disabled = Template.bind({});
exports.Disabled.args = {
    disabled: true,
};
