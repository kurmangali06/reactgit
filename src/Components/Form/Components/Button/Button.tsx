import React, { FC } from 'react';
import ButtonUI from '@mui/material/Button';

interface ButtonProps {
  disabled: boolean,
  onButtonClick?: () => void
}

export const Button: FC<ButtonProps> = ({ disabled, onButtonClick }) => (
  <ButtonUI
    type="submit"
    variant="outlined"
    disabled={disabled}
    onClick={onButtonClick}
  >
    click
  </ButtonUI>
);
