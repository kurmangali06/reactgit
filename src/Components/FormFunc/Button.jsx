
export const Button = (props) => {
  return <button onClick={props.click} className='submit'>{props.name}</button>
}