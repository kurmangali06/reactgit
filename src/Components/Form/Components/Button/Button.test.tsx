import React from 'react';
import { Button } from './Button';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { waitFor } from '@storybook/testing-library';

describe('Button', () => {
  it('render component', () => {
    render(<Button disabled={false} />);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<Button disabled={false} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render component with text', () => {
    render(<Button disabled={false} />);
    expect(screen.getByText(/click/)).toBeInTheDocument();
  });

  it('render multiply components', () => {
    render(
      <>
        <Button disabled={false} />
        <Button disabled={false} />
      </>
    );
    expect(screen.queryAllByRole('button').length).toBe(2);
  });

  it('button is disabled', () => {
    render(<Button disabled />);

    expect(screen.getByText('click')).toBeDisabled();
  });

  it('button async click', async () => {
    const mockHandler = jest.fn();
    render(<Button disabled={false} onButtonClick={() => setTimeout(mockHandler, 1000)} />);

    userEvent.click(screen.getByText(/click/));

    await waitFor(() => expect(mockHandler).toHaveBeenCalledTimes(1), {
      timeout: 1100,
    });
  });
});
