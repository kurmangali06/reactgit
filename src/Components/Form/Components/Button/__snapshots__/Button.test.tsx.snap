// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="submit"
  >
    click
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
